t/t2230-job-info-list.t

```diff
@@ -979,6 +981,59 @@ test_expect_success HAVE_JQ 'list-inactive request with invalid input fails with
 	test_cmp ${name}.expected ${name}.out
 '
 
+# Following tests use invalid jobspecs, must load a more permissive validator
+
+ingest_module ()
+{
+        cmd=$1; shift
+        flux module ${cmd} job-ingest $* &&
+        flux exec -r all -x 0 flux module ${cmd} job-ingest $*
+}
+
+test_expect_success 'reload job-ingest with more permissive validator' '
+        ingest_module reload \
+                validator=${JSONSCHEMA_VALIDATOR} validator-args=--schema,${PERMISSIVE_SCHEMA}
+'
+
+test_expect_success HAVE_JQ 'create illegal jobspec with empty command array' '
+        cat hostname.json | $jq ".tasks[0].command = []" > bad_jobspec.json
+'
+
+# to avoid potential racyness, wait up to 5 seconds for job to appear
+# in job list.  note that how a jobspec is illegal can affect what
+# values below are set vs not set.
+test_expect_success HAVE_JQ 'flux job list works on job with illegal jobspec' '
+        jobid=`flux job submit bad_jobspec.json | flux job id` &&
+        fj_wait_event $jobid clean >/dev/null &&
+        i=0 &&
+        while ! flux job list --states=inactive | grep $jobid > /dev/null \
```

Comment:
Would it be hepful in general for job listing tests to add an RPC to `job-info` that blocks until a particular job id is fully loaded into memory?  Then instead of polling, a test like this one could just run:
```
echo $jobid | ${FLUX_BUILD_DIR}/t/request/rpc job-info.wait-job
```
before expecting `flux job list` to contain updated output?

Without looking at the code, I had in mind the following limitations to keep it simple:
- only intended for testing
- only one request allowed globally (EBUSY for second one)
- hang forever if job never shows up
- disconnect handler would need to handle client exit with pending request

Just wanted to throw that idea out there - I'm not sure if it would be as simple as I was picturing.

Reply:
> Would it be hepful in general for job listing tests to add an RPC to job-info that blocks until a particular job id is fully loaded into memory?

In this particular case its a little tricky b/c we want the job id loaded into memory AND a particular state to be reached.  So we can't use `flux job list-ids`.  The state being reached is necessary so we know that all of the jobspec / R / eventlog parsing has been done.  Let me atleast make a comment to that effect so there is understanding of why this is done.

Reply:
I was just thinking in general the open loop update between job manager and job info creates complications for testing.  If there were a simple way to synchronize then perhaps a lot of inscrutable shell code and maybe some unnecessary polling delays could be eliminated.  Maybe this rpc could take an optional state?

Anyway, just an idea.

Reply:
I really like this. Whenever we see looping synchronization code we should try to replace with an idea like this.

It is fine if it doesn't make it into this PR, but we should open an issue  on it in that case.